### PR TITLE
Add ability to output specific data

### LINKED ISSUE #26979: Support specific data output in `RayTracingMeshOutput`
## Reason
We currently only support the output of all data via `output_data` and `output_aux_data`. We should support outputting specific data names.

## Design
Add params `output_data_names` and `output_aux_data_names` to support outputting specific data.

## Impact
New capability.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #26979

@cticenhour - this is in support of FENIX

### CODE REVIEW

**Comment by loganharbour** [comment]
FYI @gsgall - adds `output_data_names` and `output_aux_data_names`

**Comment by cticenhour** [comment]
Great - will make a note to come back and check the CI results later today when I have time for the review. Thanks!

**Comment by loganharbour** [comment]
Good to go now.

**Comment by loganharbour** [comment]
OpenMPI failure is unrelated... sigh

File: modules/ray_tracing/test/tests/outputs/ray_tracing_mesh_output/tests [hunk 1]
```diff
@@ -176,25 +207,65 @@
...
       type = RunException
       input = 'ray_mesh_output_errors.i'
       cli_args = 'UserObjects/study/data_on_cache_traces=true
                   UserObjects/study/segments_on_cache_traces=false
+                  Outputs/rays/output_data=true
                   Outputs/rays/output_data_nodal=true'
       expect_err = "Not supported when study segments_on_cache_traces = false"
 
-      detail = 'and when nodal data is requested to be output but segment-wise information is not cached.'
+      detail = 'when nodal data is requested to be output but segment-wise information is not cached,'
```
Comment (cticenhour) [inline modules/ray_tracing/test/tests/outputs/ray_tracing_mesh_output/tests:228]:
```suggestion
      detail = 'nodal data is requested to be output but segment-wise information is not cached,'
```

File: modules/ray_tracing/test/tests/outputs/ray_tracing_mesh_output/tests [hunk 2]
```diff
@@ -176,25 +207,65 @@
...
+    []
+    [data_all_and_specific]
+      type = RunException
+      input = 'ray_mesh_output_errors.i'
+      cli_args = 'UserObjects/study/data_on_cache_traces=true
+                  Outputs/rays/output_data=true
+                  Outputs/rays/output_data_names=unused'
+      expect_err = "Cannot be used in addition to 'output_data_names'"
+
+      detail = 'when all data and specific data are both requested,'
```
Comment (cticenhour) [inline modules/ray_tracing/test/tests/outputs/ray_tracing_mesh_output/tests:238]:
```suggestion
      detail = 'all data and specific data are both requested,'
```

File: modules/ray_tracing/test/tests/outputs/ray_tracing_mesh_output/tests [hunk 3]
```diff
@@ -176,25 +207,65 @@
...
+    []
+    [aux_data_all_and_specific]
+      type = RunException
+      input = 'ray_mesh_output_errors.i'
+      cli_args = 'UserObjects/study/aux_data_on_cache_traces=true
+                  Outputs/rays/output_aux_data=true
+                  Outputs/rays/output_aux_data_names=unused'
+      expect_err = "Cannot be used in addition to 'output_aux_data_names'"
+
+      detail = 'when all aux data and specific aux data are both requested'
```
Comment (cticenhour) [inline modules/ray_tracing/test/tests/outputs/ray_tracing_mesh_output/tests:248]:
```suggestion
      detail = 'all aux data and specific aux data are both requested'
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
A few small things. Sorry for the delay. Nearly finished this and then got distracted and didn't come back for a while.

**Comment by loganharbour** [comment]
OpenMPI failure is unrelated.

**Review by cticenhour — APPROVED** [review]
OpenMPI failure is unrelated. 

@gsgall - tagging you for awareness that this is merged in.

**Pull request merged**